codec-http2/src/main/java/io/netty/handler/codec/http2/PriorityStreamByteDistributor.java

```diff
@@ -379,46 +383,33 @@ long unallocatedStreamableBytesForTree() {
     /**
      * A connection stream visitor that delegates to the user provided visitor.
      */
-    private class WriteVisitor implements Http2StreamVisitor {
-        Writer writer;
-        RuntimeException error;
-
-        void writeAllocatedBytes(Writer writer) {
-            try {
-                this.writer = writer;
-                try {
-                    connection.forEachActiveStream(this);
-                } catch (Http2Exception e) {
-                    // Should never happen since the visitor doesn't throw.
-                    throw new IllegalStateException(e);
-                }
-
-                // If an error was caught when calling back the visitor, throw it now.
-                if (error != null) {
-                    throw error;
-                }
-            } finally {
-                error = null;
-            }
+    private final class WriteVisitor implements Http2StreamVisitor {
+        private final Deque<Writer> writers = new ArrayDeque<Writer>(1);
+        private Writer writer;
+
+        void writeAllocatedBytes(Writer writer) throws Http2Exception {
+            this.writer = writer;
+            writers.push(writer);
+            connection.forEachActiveStream(this);
+            writers.pop();
+            this.writer = writers.peek();
         }
 
         @Override
-        public boolean visit(Http2Stream stream) {
+        public boolean visit(Http2Stream stream) throws Http2Exception {
             PriorityState state = state(stream);
-            try {
-                int allocated = state.allocated;
+            int allocated = state.allocated;
 
-                // Unallocate all bytes for this stream.
-                state.resetAllocated();
+            // Unallocate all bytes for this stream.
+            state.resetAllocated();
```

Comment:
The goal of the previous code was to always ensure that we reset allocated on all streams.  I guess if we consider an exception during the write to be a `connectionError` it doesn't matter much.